R/vt_template.R

```diff
@@ -0,0 +1,50 @@
+#' templating function
+#'
+#' Access valtools templates and create modified version based on the input data
+#'
+#' @param path output path to save rendered template to
+#' @param template template to use from valtools/inst/templates
+#' @param data named list or environment with variables that will be used during rendering
+#'
+#' @importFrom whisker whisker.render
+#' @importFrom rlang abort
+#'
+#' @noRd
+#'
+render_template <- function(template, output = template, data = list()){
+
+  template_path <- file.path(
+    system.file("templates", package = "valtools"),
+    template)
+
+  if(!file.exists(template_path)){
+    abort(paste0(c("Error during creation of template `",template,"`. Error: ",
+                   e, sep = "\n")),
+          class = "vt.template_exist_fail")
+  }
+
+  template_text <- readLines(template_path)
+
+  tryCatch({
+
+    output_text <- whisker.render(
+      template = template_text,
+      data = data)
+
+    file_con <- file(output)
+
+    writeLines(
+      output_text,
+      con = file_con
```

Comment:
I'm thinking we might want to have a `finally{close(file_con)}` for all of the `writeLines`. Thoughts?

Reply:
Good point. Yes, add that in

Reply:
Looks good. I also added this as issue #30. I'll submit a PR for that once we get everything else all merged in.